packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -69,222 +107,325 @@ abstract class FlutterGoldenFileComparator extends GoldenFileComparator {
+///  * [GoldenFileComparator], the abstract class that
+///    [FlutterGoldenFileComparator] implements.
+///  * [FlutterSkiaGoldFileComparator], another
+///    [FlutterGoldenFileComparator] that uploads tests to the Skia Gold
+///    dashboard.
+///  * [FlutterPreSubmitFileComparator], another
+///    [FlutterGoldenFileComparator] that tests golden images before changes are
+///    merged into the master branch.
+class FlutterLocalFileComparator extends FlutterGoldenFileComparator with LocalComparisonOutput {
+  /// Creates a [FlutterLocalFileComparator] that will test golden file
+  /// images against baselines requested from Flutter Gold.
+  ///
+  /// The [fs] and [platform] parameters useful in tests, where the default file
+  /// system and platform can be replaced by mock instances.
+  FlutterLocalFileComparator(
+    final Uri basedir,
+    final SkiaGoldClient skiaClient, {
+    final FileSystem fs = const LocalFileSystem(),
+    final Platform platform = const LocalPlatform(),
+  }) : super(
+    basedir,
+    skiaClient,
+    fs: fs,
+    platform: platform,
+  );
+
+  /// Creates a new [FlutterLocalFileComparator] that mirrors the
+  /// relative path resolution of the default [goldenFileComparator].
+  ///
+  /// The [goldens] and [defaultComparator] parameters are visible for testing
+  /// purposes only.
+  static Future<FlutterGoldenFileComparator> fromDefaultComparator(
+    final Platform platform, {
+    SkiaGoldClient goldens,
     LocalFileComparator defaultComparator,
-  }) {
+  }) async {
+
     defaultComparator ??= goldenFileComparator;
-    return FlutterSkippingGoldenFileComparator(defaultComparator.basedir);
+    final Directory baseDirectory = FlutterGoldenFileComparator.getBaseDirectory(
+      defaultComparator,
+      platform,
+    );
+    if (!baseDirectory.existsSync())
+      baseDirectory.createSync(recursive: true);
+
+    goldens ??= SkiaGoldClient(baseDirectory);
+    await goldens.getExpectations();
+
+    return FlutterLocalFileComparator(baseDirectory.uri, goldens);
   }
 
   @override
   Future<bool> compare(Uint8List imageBytes, Uri golden) async {
-    print('Skipping "$golden" test : Skia Gold is not available in this testing '
-      'environment and flutter/goldens repository comparison is only available '
-      'on Linux machines.'
-    );
-    return true;
-  }
+    golden = addPrefix(golden);
+    final String testName = skiaClient.cleanTestName(golden.path);
+    final Map<String, dynamic> testExpectations = skiaClient.expectations[testName];
 
-  @override
-  Future<void> update(Uri golden, Uint8List imageBytes) => null;
+    if (testExpectations == null) {
+      // There is no baseline for this test
+      print('No expectations provided by Skia Gold for test: $golden. '
+        'This may be a new test. If this is an unexpected result, check'
+        ' https://flutter-gold.skia.org.\n'
+        'Image output found at $basedir'
```

Comment:
Should this include a call to action to double check that the output found there is what's expected?